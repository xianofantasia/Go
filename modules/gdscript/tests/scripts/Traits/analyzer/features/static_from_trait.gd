trait SomeTrait:
	static var some_static_var = 0
	static var other_static_var = 0
	static var third_static_var = 0

	static func some_static_func():
		print("some static func")

	static func other_static_func():
		print("other static func")

	static func third_static_func():
		print("third static func")

class SomeClass:
	uses SomeTrait

	# Overridden static variable
	var other_static_var = 1 # using 'static' keyword is optional
	static var third_static_var = 1

	# Overridden static function
	func other_static_func(): # using 'static' keyword is optional
		print("overridden other static func")

	static func third_static_func():
		print("overridden third static func")

class ThirdClass:
	uses SomeTrait

func test():
	print(SomeClass.some_static_var)
	print(SomeClass.other_static_var)
	print(SomeClass.third_static_var)
	print(ThirdClass.some_static_var)
	SomeClass.some_static_func()
	SomeClass.other_static_func()
	SomeClass.third_static_func()
	ThirdClass.some_static_func()
	print("ok")
