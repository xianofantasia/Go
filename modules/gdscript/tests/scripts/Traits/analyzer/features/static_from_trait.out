GDTEST_OK
~~ WARNING at line 19: (UNUSED_STATIC_OVERRIDING_TRAIT) Overridden static without using "static" keyword.
~~ WARNING at line 23: (UNUSED_STATIC_OVERRIDING_TRAIT) Overridden static without using "static" keyword.
0
1
1
0
some static func
overridden other static func
overridden third static func
some static func
ok
